sdk/core/src/main/java/com/klaviyo/coresdk/networking/UserInfo.kt

```diff
@@ -6,9 +6,70 @@ package com.klaviyo.coresdk.networking
  * Stores information on the currently active user
  */
 internal object UserInfo {
+    var externalId: String = ""
     var email: String = ""
+    var phoneNumber: String = ""
+    // TODO should anon ID be here with all the other identifiers
+
+    fun hasExternalId(): Boolean {
+        return externalId.isNotEmpty()
+    }
 
     fun hasEmail(): Boolean {
         return email.isNotEmpty()
     }
+
+    fun hasPhoneNumber(): Boolean {
+        return phoneNumber.isNotEmpty()
+    }
+
+    fun reset() {
+        externalId = ""
+        email = ""
+        phoneNumber = ""
+    }
+
+    /**
+     * Construct a [KlaviyoCustomerProperties] object containing current UserInfo identifiers
+     *
+     * @return
+     */
+    fun getAsCustomerProperties(): KlaviyoCustomerProperties {
+        return KlaviyoCustomerProperties().also {
+            setCustomerProperties(it)
+        }
+    }
+
+    /**
+     * Apply all identifiers from UserInfo to a [KlaviyoCustomerProperties] object
+     *
+     * @param properties
+     */
+    private fun setCustomerProperties(properties: KlaviyoCustomerProperties) {
+        if (hasExternalId()) {
+            properties.setIdentifier(externalId)
+        }
+
+        if (hasEmail()) {
+            properties.setEmail(email)
+        }
+
+        if (hasPhoneNumber()) {
+            properties.setPhoneNumber(phoneNumber)
+        }
+    }
+
+    /**
+     * Two-way merge of a [KlaviyoCustomerProperties] object with UserInfo
+     * Identifiers present on the incoming object will be applied to UserInfo
+     * Any other identifiers will be added to the properties object from UserInfo
+     *
+     * @param properties
+     */
+    fun mergeCustomerProperties(properties: KlaviyoCustomerProperties) {
```

Comment:
@jordan-griffin following on our thread about `setProfile`, here's an idea to merge the incoming profile properties _into_ state and update it with other identifiers _from_ state